### ISSUE TITLE
Importing from LDAP hangs

### ISSUE THREAD

**Issue by SGVubmluZ0cK** [zammad/zammad#1008]
### Infos:

* Zammad version: 1.5
* Zammad installed from deb-package
* Operating system: Ubuntu Server 16.04
* Browser + version:  Firefox 53

### Expected behavior:

* Import is done in reasonable time 

### Actual behavior:

* The Import is stuck since >24h
![zammadldap](https://cloud.githubusercontent.com/assets/28089775/25477927/7e476dec-2b3f-11e7-8ce0-2f13988a71dc.PNG)

**rails admin console output** 

```
irb(main):004:0> Setting.get('import_backends')
=> ["Import::Ldap"]
```

```
irb(main):005:0> ImportJob.last.attributes.except('payload').inspect
=> "{\"id\"=>55, \"name\"=>\"Import::Ldap\", \"dry_run\"=>false, \"result\"=>{\"skipped\"=>2, \"created\"=>2, \"updated\"=>0, \"unchanged\"=>496, \"failed\"=>0, \"sum\"=>2403, \"role_ids\"=>{3=>{\"created\"=>2, \"updated\"=>0, \"unchanged\"=>493, \"failed\"=>0}, 2=>{\"created\"=>0, \"updated\"=>0, \"unchanged\"=>3, \"failed\"=>0}, 1=>{\"created\"=>0, \"updated\"=>0, \"unchanged\"=>1, \"failed\"=>0}}}, \"started_at\"=>Wed, 26 Apr 2017 09:57:42 UTC +00:00, \"finished_at\"=>nil, \"created_at\"=>Wed, 26 Apr 2017 09:57:42 UTC +00:00, \"updated_at\"=>Wed, 26 Apr 2017 10:00:11 UTC +00:00}"
```

```
irb(main):006:0> ImportJob.where(name: 'Import::Ldap', dry_run: false).order(created_at: :desc).limit(1).first.attributes.except('payload').inspect
=> "{\"id\"=>55, \"name\"=>\"Import::Ldap\", \"dry_run\"=>false, \"result\"=>{\"skipped\"=>2, \"created\"=>2, \"updated\"=>0, \"unchanged\"=>496, \"failed\"=>0, \"sum\"=>2403, \"role_ids\"=>{3=>{\"created\"=>2, \"updated\"=>0, \"unchanged\"=>493, \"failed\"=>0}, 2=>{\"created\"=>0, \"updated\"=>0, \"unchanged\"=>3, \"failed\"=>0}, 1=>{\"created\"=>0, \"updated\"=>0, \"unchanged\"=>1, \"failed\"=>0}}}, \"started_at\"=>Wed, 26 Apr 2017 09:57:42 UTC +00:00, \"finished_at\"=>nil, \"created_at\"=>Wed, 26 Apr 2017 09:57:42 UTC +00:00, \"updated_at\"=>Wed, 26 Apr 2017 10:00:11 UTC +00:00}"
```

### Steps to reproduce the behavior:

*

**Comment by thorsteneckel** [comment]
Hi @SGVubmluZ0cK - could you please provide the `production.log` file from your `log/` directory? If you don't want to post it here you can reach us at support [at] zammad.com

**Comment by thorsteneckel** [comment]
Hi @SGVubmluZ0cK - thanks for the provided `production.log`! As I wrote you before I couldn't find anything that could cause this. There were no error messages either. I assume that the Scheduler was stopped unexpectedly. But I'm not totally sure.

However. Please execute the following commands to restart a new import job:

Stop the current run with:  `ImportJob.last.update_attribute('finished_at', Time.zone.now)`

Start a new one: `ImportJob.create(name: 'Import::Ldap', payload: Setting.get('ldap_config')).delay.start`

Please let me know how it went. If there are still issues please provide the log file again. There should be two scheduler log files in the log directory too. `scheduler_err.log` and `scheduler_out.log` please send them with your mail then.

Good luck 🤞

**Comment by thorsteneckel** [comment]
Hi @SGVubmluZ0cK - any updates on this? Could you please provide the name of the LDAP software you are using? Is it possible that the there is a limit on the maximum delivered entries for a search request?

**Comment by SGVubmluZ0cK** [comment]
Hi @thorsteneckel 
The LDAP-Connection seems to be ok. Sometimes the Sync-Process runs for a very long time. 
The LDAP that we are using is a Windows Active Directory with fairly standard settings. Using other servers with LDAP we never experienced any limitations, as you asked me about information on this. The scheduler_err.log and scheduler_out.log are both empty. 

What would be the effect of disabling LDAP integration. Would all users and agents be deleted?

Thank you for your advice and help.

**Comment by kmclea16** [comment]
Just noticed our LDAP sync seems to be hanging too. As far as I can see, all users were originally imported so they can still log in. I haven't had a chance yet to restart the import job. We are also using Active Directory.

<img width="649" alt="screen" src="https://cloud.githubusercontent.com/assets/28083040/25697065/eb978ffe-30fc-11e7-8198-3212e1f984bb.PNG">

**Comment by thorsteneckel** [comment]
Hi @kmclea16 can you please use the [Zammad rails console](https://docs.zammad.org/en/latest/admin-console.html#start-zammad-s-rails-console) and execute the command `ImportJob.last.attributes.except('payload').inspect` and provide the output like above? How is the load of your system when the import is running and when it isn't running?

@SGVubmluZ0cK this is really strange 🤔 I saw your other issue #1036 maybe there is something deeper wrong. Is the load only high when the LDAP import is running or a general state? 

A question to the both of you:
How long did it take to count the entries? Just a rough estimation is enough here.

**Comment by SGVubmluZ0cK** [comment]
As mentioned in the issue [#1036](https://github.com/zammad/zammad/issues/1036) the load keeps high even when the import is done. The last complete finished at 8:05. Currently there is a sync job running since 9:06.

**Comment by kmclea16** [comment]
Hi @thorsteneckel 

Here's the output:

`irb(main):001:0> ImportJob.last.attributes.except('payload').inspect
=> "{\"id\"=>19, \"name\"=>\"Import::Ldap\", \"dry_run\"=>false, \"result\"=>{\"skipped\"=>5, \"created\"=>0, \"updated\"=>2, \"unchanged\"=>293, \"failed\"=>0, \"sum\"=>7958, \"role_ids\"=>{3=>{\"created\"=>0, \"updated\"=>2, \"unchanged\"=>293, \"failed\"=>0}}}, \"started_at\"=>Sat, 22 Apr 2017 18:02:30 UTC +00:00, \"finished_at\"=>nil, \"created_at\"=>Sat, 22 Apr 2017 18:02:29 UTC +00:00, \"updated_at\"=>Sat, 22 Apr 2017 18:03:29 UTC +00:00}"
irb(main):002:0>`

For load, do you mean system performance? It doesn't seem to be high

root@zammad:~# mpstat
Linux 4.4.0-75-generic (zammad)         04/05/17        _x86_64_        (8 CPU)
21:04:01     CPU    %usr   %nice    %sys %iowait    %irq   %soft  %steal  %guest  %gnice   %idle
21:04:01     all    0.65    0.00    0.07    0.03    0.00    0.01    0.00    0.00    0.00   99.24

**Comment by thorsteneckel** [comment]
Thanks @kmclea16. Looks good to me. Can you send me what `Delayed::Job.all` return when the LDAP import hangs?

**Comment by Jonske** [comment]
I have the same problem. I'm on Centos7 and for the first 2-3 weeks it worked perfectly fine for me. But since last friday, the ldap-sync hangs. It works when i manually start a new import job. But it would be good if it worked like before.

Another problem that i'm running into is, that when the ldap-sync is finished it says that 30 users got skipped. I already checked their attributes, but they are identical to the other users that got synchronized. It seems very random and it would be cool if they get synchronized aswell.

**Comment by thorsteneckel** [comment]
Hi @Jonske! Regarding your first issue: 
Have you tried the statements I've posted above in the issue? If so, what's the output? I'm currently adding some debug outputs to trace this issue. Currently I can't tell where and why the import stops out of the sudden without logging an error somewhere 🤔

Regarding the skipped users: There is an upcoming change that is currently in QA. There was an error detecting changed attributes. This might lead to the not syncing. Generally all users where no attributes where changed are skipped so maybe there were no changes to them?

**Comment by Jonske** [comment]
Hey @thorsteneckel! Thank you for your fast answer. Here is the output:
`irb(main):001:0> ImportJob.last.attributes.except('payload').inspect
=> "{\"id\"=>182, \"name\"=>\"Import::Ldap\", \"dry_run\"=>false, \"result\"=>{\"skipped\"=>33, \"created\"=>1, \"updated\"=>267, \"unchanged\"=>493, \"failed\"=>0, \"sum\"=>794, \"role_ids\"=>{3=>{\"created\"=>1, \"updated\"=>267, \"unchanged\"=>487, \"failed\"=>0}, 2=>{\"created\"=>0, \"updated\"=>0, \"unchanged\"=>6, \"failed\"=>0}}}, \"started_at\"=>Mon, 08 May 2017 13:28:19 UTC +00:00, \"finished_at\"=>Mon, 08 May 2017 13:28:51 UTC +00:00, \"created_at\"=>Mon, 08 May 2017 13:28:18 UTC +00:00, \"updated_at\"=>Mon, 08 May 2017 13:28:51 UTC +00:00}"`

To the skipped users issue: I don't know excatly what you mean, my english isn't the best, sorry :/ I'll try to explain it a little more detail:
I created a new user in my Active Directory today, and the ldap-sync worked fine. Then i have some users that i created 1 year ago at the same time. They are all identical, except for the names and e-mail address of course. 2 of them get synchronized, but the other 3 don't. The user-profiles are all used everyday, so it seems random to me.

But maybe i didn't understand what you meant before.

**Comment by thorsteneckel** [comment]
Hi @Jonske - you're welcome 👍 

The output looks fine to me. The job started and finished for 794 users within less than a minute 🚀 

Here is a list why a user might get skipped:
- When there is no `login`
- When there are no other attributes except `login` and `dn`
- When no role assignment can be found and skipping is configured (starting from the next release on)

Can you check if the skipped users apply to one of these skip criterias? I'll add some output for skipped users too so it will be easier to debug this.

If the LDAP sync hangs again please re-execute the command and post the output here. Thanks!

**Comment by kmclea16** [comment]
Our LDAP sync has been running perfectly fine (for about a week) since I restarted the ImportJob 👍

**Comment by Jonske** [comment]
I restarted the ImportJob today and it says:

`irb(main):002:0> ImportJob.create(name: 'Import::ldap', payload:
irb(main):003:1* Setting.get('ldap_config')).delay.start
Setting.cache_valid?: cache still valid, 26332125/26332125
=> #<Delayed::Backend::ActiveRecord::Job id: 1456, priority: 0, attempts: 0, handler: "--- !ruby/object:Delayed::PerformableMethod\nobject...", last_error: nil, run_at: "2017-05-09 05:50:41", locked_at: nil, failed_at: nil, locked_by: nil, queue: nil, created_at: "2017-05-09 05:50:41", updated_at: "2017-05-09 05:50:41">`

If I look at the log it says:

`irb(main):004:0> ImportJob.last.attributes.except('payload').inspect
=> "{\"id\"=>183, \"name\"=>\"Import::ldap\", \"dry_run\"=>false, \"result\"=>{\"error\"=>\"wrong constant name ldap\"}, \"started_at\"=>Tue, 09 May 2017 05:50:48 UTC +00:00, \"finished_at\"=>Tue, 09 May 2017 05:50:49 UTC +00:00, \"created_at\"=>Tue, 09 May 2017 05:50:41 UTC +00:00, \"updated_at\"=>Tue, 09 May 2017 05:50:49 UTC +00:00}"`

Any idea why it says "wrong constant name ldap"? I didn't change anything in my ldap settings. 

To the skipped users:
They all have more attributes than 'login' and 'dn'. I didn't assign a role to the other users, since they all get the standard role 'customer'. I will try to create a group in my AD and assign it to the skipped users. Maybe that will help.

**Comment by thorsteneckel** [comment]
@Jonske - Nice! That's an error message I can work with. Can you please check your `log/production.log` file for that message? There should be more information if your search for the text `wrong constant name ldap`. Can you tell me a bit more about your system? Which OS? Which Zammad installation source? Was it an job you started or did Zammad started it automatically? Thanks!

@kmclea16 - Thanks for the feedback. If there's anything just let me know.

**Comment by Jonske** [comment]
@thorsteneckel  I'm running CentOS 7 on a virtual machine (VMWare). I installed Zammad via RPM. I have more than enough storage and RAM on my VM.

I started the job manually. It only works when i go to **/opt/zammad/app/controllers/integration** and then remove the "ldap_controller.rb" and the "sipgate_controller.rb". After that, I have to restart zammad and then it lets me change the ldap-settings on the web-interface again. I put both files back into the folder and then sync works perfect. It doesn't even work when I manually start the import in zammad rails.

I looked into the production.log, but i can't find anything under "wrong constant name ldap", but I found something different. I always see the proxy-IP when i look for the word "ldap". Obviously, I can't send you my entire log-file, because it would be to long, but i copied some lines:

I, [2017-05-09T09:55:07.731478 #22392]  INFO -- : Started GET "/api/v1/integration/ldap/job_start?_=1494316264588" for 10.76.0.7 at 2017-05-09 09:55:07 +0200
I, [2017-05-09T09:55:07.734736 #22392]  INFO -- : Processing by Integration::LdapController#job_start_index as JSON
I, [2017-05-09T09:55:07.734811 #22392]  INFO -- :   Parameters: {"_"=>"1494316264588", "ldap"=>{}}
I, [2017-05-09T09:55:07.745712 #22392]  INFO -- : Started GET "/api/v1/integration/ldap/job_try?finished=false&_=1494316264589" for 10.76.0.7 at 2017-05-09 09:55:07 +0200
I, [2017-05-09T09:55:07.747937 #22392]  INFO -- : Processing by Integration::LdapController#job_try_index as JSON
I, [2017-05-09T09:55:07.748002 #22392]  INFO -- :   Parameters: {"finished"=>"false", "_"=>"1494316264589", "ldap"=>{}}
I, [2017-05-09T09:55:07.758996 #22392]  INFO -- : Completed 200 OK in 11ms (Views: 0.1ms | ActiveRecord: 3.4ms)

It looks like zammad is trying to sync with the wrong IP or am I wrong? Is there a way to reset the ldap configuration? Like a ldap-configuration file? 

I hope that helps.

**Comment by thorsteneckel** [comment]
Thanks for the comprehensive feedback! I'll set up a VM as well and try to reproduce it.

The workaround with removing the controllers is 😨 That should definitively not be needed. But respect for finding a way that make it works 🤓 

Can you please check your `log/scheduler_err.log` for messages like `log writing failed. closed stream`. We had an issue with the scheduler losing the file handle to the `log/production.log` making it not possible to log from scheduler/background jobs like the LDAP import jobs are.
If there are these messages please update to the latest stable version via your package manager. The latest version includes [a fix](https://github.com/zammad/zammad/commit/22e81f6ba3f431ea34891dc02aea488bca14d54a) for this. 

Regarding the IP: This is just Rails logging from where the request came. So it's not the IP of the LDAP we are requesting. These requests are polling the current state of the LDAP or trying to do so...

I may already have a solution but I'm trying to reproduce it first. I'll keep you updated. Thanks so far!

**Comment by Jonske** [comment]
My "scheduler_err.log" files had the "log writing failed. closed stream." message. I will try to update zammad and see if that helps.

Thank you for your help so far.

**Comment by thorsteneckel** [comment]
Bad news: I wasn't able to reproduce the error in the latest version :/ I think it has something to do with the user and the actions performed before starting the LDAP but I have no idea what exactly.

Looking forward to get the log entry after you updated to the latest version 🤞

**Comment by Jonske** [comment]
Bad news from me too: I updated to the latest version, but it's still the same :( 
I installed zammad via RPM, so "sudo yum update zammad" should be enough, right? 
On my web-interface, it says that I'm on version 1.5.x.
Is 1.6 still in beta or why didn't it update on 1.6?

I really don't know what to do now anymore :/

**Comment by Jonske** [comment]
This is the only error i can find in the production.log. Sorry if it's to long.

F, [2017-05-09T13:22:55.611958 #5954] FATAL -- : 
ActionController::RoutingError (uninitialized constant Integration::LdapController):
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/inflector/methods.rb:263:in `const_get'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/inflector/methods.rb:263:in `block in constantize'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/inflector/methods.rb:259:in `each'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/inflector/methods.rb:259:in `inject'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/inflector/methods.rb:259:in `constantize'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/routing/route_set.rb:70:in `controller_reference'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/routing/route_set.rb:60:in `controller'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/routing/route_set.rb:39:in `serve'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/journey/router.rb:43:in `block in serve'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/journey/router.rb:30:in `each'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/journey/router.rb:30:in `serve'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/routing/route_set.rb:817:in `call'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:186:in `call!'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:164:in `call'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:186:in `call!'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:164:in `call'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:186:in `call!'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:164:in `call'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:186:in `call!'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:164:in `call'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:186:in `call!'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:164:in `call'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:186:in `call!'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:164:in `call'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:186:in `call!'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/strategy.rb:164:in `call'
  vendor/bundle/ruby/2.3.0/gems/omniauth-1.3.1/lib/omniauth/builder.rb:63:in `call'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/etag.rb:24:in `call'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/conditionalget.rb:25:in `call'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/head.rb:13:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/flash.rb:260:in `call'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:225:in `context'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/session/abstract/id.rb:220:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/cookies.rb:560:in `call'
  vendor/bundle/ruby/2.3.0/gems/activerecord-4.2.7.1/lib/active_record/query_cache.rb:36:in `call'
  vendor/bundle/ruby/2.3.0/gems/activerecord-4.2.7.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:653:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:88:in `__run_callbacks__'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/callbacks.rb:81:in `run_callbacks'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/rack/logger.rb:38:in `call_app'
  vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/rack/logger.rb:20:in `block in call'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/tagged_logging.rb:68:in `block in tagged'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/tagged_logging.rb:26:in `tagged'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/tagged_logging.rb:68:in `tagged'
  vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/rack/logger.rb:20:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/request_id.rb:21:in `call'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/methodoverride.rb:22:in `call'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/runtime.rb:18:in `call'
  vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.7.1/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.7.1/lib/action_dispatch/middleware/static.rb:120:in `call'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/sendfile.rb:113:in `call'
  vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/engine.rb:518:in `call'
  vendor/bundle/ruby/2.3.0/gems/railties-4.2.7.1/lib/rails/application.rb:165:in `call'
  vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/content_length.rb:15:in `call'
  vendor/bundle/ruby/2.3.0/gems/puma-3.6.0/lib/puma/configuration.rb:225:in `call'
  vendor/bundle/ruby/2.3.0/gems/puma-3.6.0/lib/puma/server.rb:578:in `handle_request'
  vendor/bundle/ruby/2.3.0/gems/puma-3.6.0/lib/puma/server.rb:415:in `process_client'
  vendor/bundle/ruby/2.3.0/gems/puma-3.6.0/lib/puma/server.rb:275:in `block in run'
  vendor/bundle/ruby/2.3.0/gems/puma-3.6.0/lib/puma/thread_pool.rb:116:in `block in spawn_thread'

**Comment by thorsteneckel** [comment]
Thanks for your fast feedback. Yes the package is updated since it's just a patch the version doesn't change. Zammad 1.6 will be released next week. 

The error you posted is caused by removing the `ldaps_controller.rb` like you described some posts earlier. Rails is missing that file and throws an error. So could you put it back, restart your webservice and check the logs for the previous `wrong constant name ldap` error?

**Comment by Jonske** [comment]
I can't find anything in the production.log. 
I think it's because all my scheduler_err.logs are filled with "log writing failed. closed stream"

**Comment by thorsteneckel** [comment]
HI @Jonske - please update to the latest version of Zammad 1.5. The following changes are included:

- Logging of skipped entries with their data (helps to analyze why)
- Fix of the `log writing failed. closed stream`
- Several other fixes as described in issue #981

In the meantime I was able to reproduce the issue of stuck LDAP imports by stopping/restarting the Zammad worker/scheduler while an import was running. However there is not fix for that situation yet but I'm working on it.

Glad to get feedback.

**Comment by thorsteneckel** [comment]
Hello together! This issue is fixed with the commits above. I could reproduce the restarting or stopping of the Scheduler/Zammad worker was causing the LDAP import to stuck. It won't continue or restart. This is fixed 🚀 

Please feel free to update your Zammad installation in about 45 minutes. The new version should then be available. Closing now. Thanks for all the support!
